### ISSUE TITLE
A simple bug that has existed for more than ten years for Cursor Shapes

### ISSUE THREAD

**Issue by ch1220** [PowerShell/PowerShell#12496]
# Steps to reproduce
```powershell 
powershell title bar right click -> properties -> select cusor shape other than legacy style
```
# Expected behavior
```powershell 
**cursor shapes** 
- Legacy Style 
- Underscore
- Vertical Bar
- Empty Box
- Solid Box
The above cursor shapes can be switched at will.
```
# Actual behavior
```powershell 
only legacy style is valid
```
# Environment data
```powershell 
System environment : win7 or win8 or win10
**powershell 1.0 , 2.0 ......7.0   and 7.1.0-preview.2 All of the above versions have this problem.**
Name                           Value
----                           -----
PSVersion                      7.1.0-preview.2
PSEdition                      Core
GitCommitId                    7.1.0-preview.2
OS                             Microsoft Windows 10.0.19041
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by mklement0** [comment]
Please follow the issue template when creating issues, so that we can understand and reproduce your problem, and know what versions / platforms it applies to.

**Comment by fMichaleczek** [comment]
This issue is related to PSReadline :
https://github.com/PowerShell/PSReadLine/issues/903

The bug is external, it's resolved in the Terminal Repository :
https://github.com/microsoft/terminal/issues/4124

**Comment by ch1220** [comment]
> template
Detailed questions have been updated

**Comment by fMichaleczek** [comment]
@ch1220 I give you the answer, you should closed. If you have more questions, ask Microsoft/Terminal.

**Comment by 8192bits** [comment]
For anyone still hitting this in 2026 on conhost with PowerShell 5.1 or 7.x,
I patched the PSReadLine 2.4.5 DLL directly to fix this by re-emitting
the ANSI cursor sequence in ReallyRender() after CursorVisible is restored.

[CursorShape](https://github.com/8192bits/CursorShape)

**No recompilation needed. drop the DLL, set up $PROFILE, done.**
**Providing persistent cursor shape.**

Feedback welcome.
